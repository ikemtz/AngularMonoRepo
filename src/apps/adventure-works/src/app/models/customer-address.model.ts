/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
import { IAddress } from './address.model';
import { ICustomer } from './customer.model';

export interface ICustomerAddress {
  id?: string;
  customerId?: string;
  addressId?: string;
  addressType?: string;
  address?: IAddress;
  customer?: ICustomer;
}
