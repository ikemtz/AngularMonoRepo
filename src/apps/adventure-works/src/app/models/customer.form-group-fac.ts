/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export function CustomerFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    nameStyle: new FormControl('', Validators.required),
    title: new FormControl('', Validators.maxLength(8)),
    firstName: new FormControl('', Validators.compose([Validators.required, Validators.maxLength(50)])),
    middleName: new FormControl('', Validators.maxLength(50)),
    lastName: new FormControl('', Validators.compose([Validators.required, Validators.maxLength(50)])),
    suffix: new FormControl('', Validators.maxLength(10)),
    companyName: new FormControl('', Validators.maxLength(128)),
    salesPerson: new FormControl('', Validators.maxLength(256)),
    emailAddress: new FormControl('', Validators.maxLength(50)),
    phone: new FormControl('', Validators.maxLength(50)),
    customerAddresses: new FormArray([]),
    saleOrders: new FormArray([]),
  });
};
