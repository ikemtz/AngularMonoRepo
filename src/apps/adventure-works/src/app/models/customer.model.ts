/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
import { ICustomerAddress } from './customer-address.model';
import { ISaleOrder } from './sale-order.model';

export interface ICustomer {
  id?: string;
  nameStyle?: boolean;
  title?: string;
  firstName?: string;
  middleName?: string;
  lastName?: string;
  suffix?: string;
  companyName?: string;
  salesPerson?: string;
  emailAddress?: string;
  phone?: string;
  customerAddresses?: ICustomerAddress[];
  saleOrders?: ISaleOrder[];
}
