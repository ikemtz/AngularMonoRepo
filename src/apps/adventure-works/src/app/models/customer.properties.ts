/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum CustomerProperties {
  ID = 'id',
  NAME_STYLE = 'nameStyle',
  TITLE = 'title',
  FIRST_NAME = 'firstName',
  MIDDLE_NAME = 'middleName',
  LAST_NAME = 'lastName',
  SUFFIX = 'suffix',
  COMPANY_NAME = 'companyName',
  SALES_PERSON = 'salesPerson',
  EMAIL_ADDRESS = 'emailAddress',
  PHONE = 'phone',
  CUSTOMER_ADDRESSES = 'customerAddresses',
  SALE_ORDERS = 'saleOrders',
}
