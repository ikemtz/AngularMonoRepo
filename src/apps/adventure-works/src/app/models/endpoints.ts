/**
 * This file is generated by the openapi-ts-generator
 * #endpoints-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum Endpoints {
  addresses = 'aw-odata/odata/v1/Addresses',
  customers = 'aw-odata/odata/v1/Customers',
  product_categories = 'aw-odata/odata/v1/ProductCategories',
  product_models = 'aw-odata/odata/v1/ProductModels',
  products = 'aw-odata/odata/v1/Products',
  sale_order_details = 'aw-odata/odata/v1/SaleOrderDetails',
  sale_orders = 'aw-odata/odata/v1/SaleOrders',
}
