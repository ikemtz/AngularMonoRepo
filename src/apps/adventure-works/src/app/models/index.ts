/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './address.form-group-fac';
export * from './address.model';
export * from './address.properties';
export * from './customer-address.form-group-fac';
export * from './customer-address.model';
export * from './customer-address.properties';
export * from './customer.form-group-fac';
export * from './customer.model';
export * from './customer.properties';
export * from './endpoints';
export * from './product-category.form-group-fac';
export * from './product-category.model';
export * from './product-category.properties';
export * from './product-model.form-group-fac';
export * from './product-model.model';
export * from './product-model.properties';
export * from './product.form-group-fac';
export * from './product.model';
export * from './product.properties';
export * from './sale-order-detail.form-group-fac';
export * from './sale-order-detail.model';
export * from './sale-order-detail.properties';
export * from './sale-order.form-group-fac';
export * from './sale-order.model';
export * from './sale-order.properties';
