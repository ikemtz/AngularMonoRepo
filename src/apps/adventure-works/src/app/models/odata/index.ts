/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './customer-address.form-group-fac';
export * from './customer-address.form';
export * from './customer-address.model';
export * from './customer-address.properties';
export * from './customer-address.test-obj-fac';
export * from './customer.form-group-fac';
export * from './customer.form';
export * from './customer.model';
export * from './customer.properties';
export * from './customer.test-obj-fac';
export * from './order-address.form-group-fac';
export * from './order-address.form';
export * from './order-address.model';
export * from './order-address.properties';
export * from './order-address.test-obj-fac';
export * from './order-line-item.form-group-fac';
export * from './order-line-item.form';
export * from './order-line-item.model';
export * from './order-line-item.properties';
export * from './order-line-item.test-obj-fac';
export * from './order-status-types.enum';
export * from './order.form-group-fac';
export * from './order.form';
export * from './order.model';
export * from './order.properties';
export * from './order.test-obj-fac';
export * from './product-category.form-group-fac';
export * from './product-category.form';
export * from './product-category.model';
export * from './product-category.properties';
export * from './product-category.test-obj-fac';
export * from './product-model.form-group-fac';
export * from './product-model.form';
export * from './product-model.model';
export * from './product-model.properties';
export * from './product-model.test-obj-fac';
export * from './product.form-group-fac';
export * from './product.form';
export * from './product.model';
export * from './product.properties';
export * from './product.test-obj-fac';
export * from './sales-agent.form-group-fac';
export * from './sales-agent.form';
export * from './sales-agent.model';
export * from './sales-agent.properties';
export * from './sales-agent.test-obj-fac';
export * from './shipping-types.enum';
