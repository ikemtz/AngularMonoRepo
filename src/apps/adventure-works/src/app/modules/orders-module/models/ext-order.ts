import { PagerSettings } from '@progress/kendo-angular-grid';
import { ODataResult, ODataState } from 'imng-kendo-odata';
import { IOrder, IOrderLineItem } from '../../../models/odata';

export interface IExtOrder extends IOrder {
  orderLineItemODataState: ODataState;
  orderLineItemOData: ODataResult<IOrderLineItem>;
  orderLineItemPagerSettings: PagerSettings | false;
}
