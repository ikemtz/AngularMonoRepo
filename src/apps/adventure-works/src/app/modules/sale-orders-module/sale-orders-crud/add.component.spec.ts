import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { createDataEntryMockFacade } from 'imng-kendo-data-entry/testing';
import { CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';

import { SaleOrderAddComponent } from './add.component';
import { SaleOrderCrudFacade } from './crud.facade';
import { ISaleOrder, SaleOrderProperties } from '../../../models';

describe('SaleOrderAddComponent', () => {
  let component: SaleOrderAddComponent;
  let fixture: ComponentFixture<SaleOrderAddComponent>;
  let facade: SaleOrderCrudFacade;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [SaleOrderAddComponent],
      imports: [ReactiveFormsModule, NoopAnimationsModule],
      providers: [{ provide: SaleOrderCrudFacade, useValue: createDataEntryMockFacade() }],
      schemas: [CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SaleOrderAddComponent);
    component = fixture.componentInstance;
    facade = TestBed.inject(SaleOrderCrudFacade);
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should save', () => {
    component.initForm();
    component.addEditForm.patchValue({
      [SaleOrderProperties.ID]: 'ID',
      [SaleOrderProperties.SALES_ORDER_ID]: 0,
      [SaleOrderProperties.REVISION_NUM]: 0,
      [SaleOrderProperties.DATE]: new Date(),
      [SaleOrderProperties.DUE_DATE]: new Date(),
      [SaleOrderProperties.SHIP_DATE]: new Date(),
      [SaleOrderProperties.STATUS]: 0,
      [SaleOrderProperties.IS_ONLINE_ORDER]: true,
      [SaleOrderProperties.NUM]: 'NUM',
      [SaleOrderProperties.PURCHASE_ORDER_NUM]: 'PURCHASE_ORDER_NUM',
      [SaleOrderProperties.ACCOUNT_NUM]: 'ACCOUNT_NUM',
      [SaleOrderProperties.CUSTOMER_ID]: 'CUSTOMER_ID',
      [SaleOrderProperties.SHIP_TO_ADDRESS_ID]: 'SHIP_TO_ADDRESS_ID',
      [SaleOrderProperties.BILL_TO_ADDRESS_ID]: 'BILL_TO_ADDRESS_ID',
      [SaleOrderProperties.SHIP_METHOD]: 'SHIP_METHOD',
      [SaleOrderProperties.CREDIT_CARD_APPROVAL_CODE]: 'CREDIT_CARD_APP',
      [SaleOrderProperties.SUB_TOTAL]: 0,
      [SaleOrderProperties.TAX_AMT]: 0,
      [SaleOrderProperties.FREIGHT]: 0,
      [SaleOrderProperties.TOTAL_DUE]: 0,
      [SaleOrderProperties.COMMENT]: '',
      [SaleOrderProperties.SHIP_TO_ADDRESS]: 'SHIP_TO_ADDRESS',
      [SaleOrderProperties.BILL_TO_ADDRESS]: 'BILL_TO_ADDRESS',
      [SaleOrderProperties.CUSTOMER]: 'CUSTOMER',
    });

    let item: ISaleOrder | undefined;
    facade.saveNewEntity = jest.fn(x => (item = x));
    facade.updateExistingEntity = jest.fn();
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    component.addEditForm = { ...component.addEditForm, valid: true } as any;
    component.save();
    expect(facade.saveNewEntity).toBeCalledTimes(1);
    expect(facade.updateExistingEntity).toBeCalledTimes(0);

    expect(item).toMatchSnapshot({
      date: expect.any(Date),
      dueDate: expect.any(Date),
      shipDate: expect.any(Date),
    });

  });

  it('should not save', () => {
    component.addEditForm = { valid: false } as never;
    component.save();
    expect(facade.saveNewEntity).toBeCalledTimes(0);
    expect(facade.updateExistingEntity).toBeCalledTimes(0);
  });

  it('should cancel', () => {
    facade.clearCurrentEntity = jest.fn();
    component.cancel();
    expect(facade.clearCurrentEntity).toBeCalledTimes(1);
  });
});
