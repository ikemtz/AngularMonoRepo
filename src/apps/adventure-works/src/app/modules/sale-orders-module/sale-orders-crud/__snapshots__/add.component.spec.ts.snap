// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaleOrderAddComponent should save 1`] = `
Object {
  "accountNum": "ACCOUNT_NUM",
  "billToAddress": "BILL_TO_ADDRESS",
  "billToAddressID": "BILL_TO_ADDRESS_ID",
  "comment": "",
  "creditCardApprovalCode": "CREDIT_CARD_APP",
  "customer": "CUSTOMER",
  "customerID": "CUSTOMER_ID",
  "date": Any<Date>,
  "dueDate": Any<Date>,
  "freight": 0,
  "id": undefined,
  "isOnlineOrder": true,
  "num": "NUM",
  "purchaseOrderNum": "PURCHASE_ORDER_NUM",
  "revisionNum": 0,
  "saleOrderDetails": Array [],
  "salesOrderId": 0,
  "shipDate": Any<Date>,
  "shipMethod": "SHIP_METHOD",
  "shipToAddress": "SHIP_TO_ADDRESS",
  "shipToAddressID": "SHIP_TO_ADDRESS_ID",
  "status": 0,
  "subTotal": 0,
  "taxAmt": 0,
  "totalDue": 0,
}
`;
